protocols.py

```diff
@@ -0,0 +1,19 @@
+from shared_types import MethodList
+
+def generate_protocol_name(class_name: str, method_name: str) -> str:  
+  return f'{class_name}_{"".join([name_component[0].upper() + name_component[1:] for name_component in method_name.split("_")])}'
+
+def write_protocols(method_list: MethodList) -> None:
```

Comment:
You should extract the inner logic that operates on a single method into its own function! Handle one and then you can apply it to any iterable collection you want.

```suggestion
def generate_protocol(fucn: Callable) -> str:
```

Reply:
@willfrey Just so I make sure I understand correctly... Are you saying to split this into two functions where the first receives a `Iterable` of callables from a file and for each one calls a second function that actually generates the protocol for each?

Reply:
I'll probably address this in another PR